core/src/main/java/com/airbnb/aerosolve/core/function/Spline.java

```diff
@@ -21,21 +21,23 @@ public Spline(float minVal, float maxVal, float [] weights) {
   
   // A resampling constructor
   public Spline(Spline other, int newBins) {
-    float[] newWeights = new float[newBins];
     if (newBins == other.numBins) {
-      for (int i = 0; i < newBins; i++) {
-        newWeights[i] = other.weights[i];
-      }
+      setupSpline(other.minVal, other.maxVal, other.weights.clone());
     } else {
-      float scale = 1.0f / (newBins - 1.0f);
-      float diff = other.maxVal - other.minVal;
-      for (int i = 0; i < newBins; i++) {
-        float t = i * scale;
-        float x = diff * t + other.minVal;
-        newWeights[i] = other.evaluate(x);
-      }
+      setupSpline(other.minVal, other.maxVal, other.newBinWeights(newBins));
+    }
+  }
+
+  public float[] newBinWeights(int newBins) {
```

Comment:
Can you add a comment for what this does, that it generates a new set of weights which mimics the shape of this spline rescaled to a new number of bins?